### PR TITLE
Add helper function to validate interface name length

### PR DESCRIPTION
**Pull request by arfeigin**
This PR is accompanied by https://github.com/sonic-net/sonic-buildimage/pull/20108 and https://github.com/sonic-net/sonic-utilities/pull/3397

Add helper function to validate that interface name length does not exceed allowed length configured as IFNAMSIZ. This will be used both by sonic-utilities CLI (config) and by cfgmgr.

### CODE REVIEW

**Comment by arfeigin** [comment]
@prsunny Can you help please reviewing this and the attached PRs?

**Review by nazariig — APPROVED** [review]

**Comment by dprital** [comment]
@prsunny , @qiluo-msft , Can you please approve this PR ?

File: common/interface.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        SWSS_LOG_ERROR("Invalid interface name %s length, it must not exceed %d characters", ifaceName.c_str(), IFNAMSIZ);
+        return false;
+    }
+    return true;
+}
+
+size_t ifaceNameMaxLen()
+{
+    return IFNAMSIZ;
```
Comment (qiluo-msft) [inline common/interface.h:21]: @liuh-80 The purpose of this PR is to share a const to multiple applications in C++/python/go. What is the best practice? any sample?

**Comment by qiluo-msft** [comment]
Do you still need this PR? if not, you can 'Close' it.

**Comment by arfeigin** [comment]
> Do you still need this PR? if not, you can 'Close' it.

I think it is good to have this PR in order to export IFNAMSIZ value for sonic-utilities config scripts, not?

File: common/interface.h (hunk 1 above)
Comment (liuh-80) [reply common/interface.h:21]: The best practice is to define a const variable, SWIG will generate a global read-only wrapper:
https://www.swig.org/Doc4.0/SWIGDocumentation.html#SWIG_nn13

**Comment by liuh-80** [comment]
@arfeigin , a workaround could be define a new variable:
const size_t IFACE_NAME_MAX_LEN = IFNAMESIZ;

**Comment by stepanblyschak** [comment]
@qiluo-msft @liuh-80 Please review https://github.com/sonic-net/sonic-swss-common/pull/931

**Pull request closed without merging**